blog/docs/articles/workflow-as-function-flow.md

```diff
@@ -0,0 +1,328 @@
+
+2. Optionally deploy `cloudevents-player` that will display all events flowing through the broker with following commands
+
+
+	```
+	kubectl apply -n knative-eventing -f - << EOF
+	apiVersion: serving.knative.dev/v1
+	kind: Service
+	metadata:
+	  name: cloudevents-player
+	spec:
+	  template:
+	    metadata:
+	      annotations:
+	        autoscaling.knative.dev/minScale: "1"
+	    spec:
+	      containers:
+	        - image: quay.io/ruben/cloudevents-player:latest
+	          env:
+	            - name: PLAYER_MODE
+	              value: KNATIVE
+	            - name: PLAYER_BROKER
+	              value: default
+	---
+	apiVersion: eventing.knative.dev/v1
+	kind: Trigger
+	metadata:
+	  name: cloudevents-player
+	  annotations:
+	    knative-eventing-injection: enabled
+	spec:
+	  broker: default
+	  subscriber:
+	    ref:
+	      apiVersion: serving.knative.dev/v1
+	      kind: Service
+	      name: cloudevents-player
+	EOF
+	```
+
+3. Get the url of the default broker use following command
+
+	```
+	kubectl get broker default
+	```
+
+4. Send request to the broker to start user registration
+
+	```
+	curl -v "http://broker-ingress.knative-eventing.svc.cluster.local/knative-eventing/default" \
+	-X POST \
+	-H "Ce-Id: 1234" \
+	-H "Ce-Specversion: 1.0" \
+	-H "Ce-Type: io.automatiko.examples.userRegistration" \
+	-H "Ce-Source: curl" \
+	-H "Content-Type: application/json" \
+	-d '{"user" : {"email" : "mike.strong@email.com",  "firstName" : "mike",  "lastName" : "strong"}}'
+	```
+
+	This will go via number of events being exchanged over the Knative broker and invoking functions defined in the workflow. It also uses Swagger Petstore REST api so in case of successful user registration it will be visible in Swagger Petstore as new user.
+
+	**NOTE** That Swagger Petstore does not have reliable storage thus it might require few get requests to be issued to see it there.
+
+#### Clean up
+
+To clean up, execute following command
+
+````
+kubectl delete -f target/functions/user-registration-1.0.0.yaml
+````
+
+
+### Build the service as container image for Google Cloud Run
+
+To be able to use the same service for Google Cloud Run there are two additional steps required
+
+1. Add dependency to `pom.xml`
+
+	````
+	<dependency>
+	  <groupId>io.automatiko.extras</groupId>
+	  <artifactId>automatiko-gcp-pubsub-sink</artifactId>
+	</dependency>
+	````
+
+2. Add two properties to `src/main/resources/application.properties` file
+
+	````
+	quarkus.automatiko.target-deployment=gcp-pubsub
+	quarkus.google.cloud.project-id=CHANGE_ME
+	````
+
+	**NOTE**: Remember to change `CHANGE_ME` to actual project id of your Google Cloud project
+
+3. Build the application with following maven command
+
+	````
+	mvnw clean package -Pcontainer-native
+	````
+
+	This will build a container with the service that is using native executable built with GraalVM so the build process will take a while but it will be really light and fast at execution.
```

Comment:
Same as earlier text, I'd edit this the same way and leave out the last bit, just make users aware it might take a while

Reply:
changed